### PR TITLE
Add tasks to install runsc for gvisor integration

### PR DESCRIPTION
**Pull request by heytrav**
What this PR does / why we need it:

This PR adds the option to install `runsc`, making it possible to create sandboxed pods using `gvisor`.

Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged): Fixes #

This PR does not fix any known issues. 

**Additional context**
The Ansible tasks implement the  [installation instructions](https://gvisor.dev/docs/user_guide/install/) on the gVisor website. To run them the flag `containerd_gvisor_runtime` must be set to `true`.

Then the user must create a `RuntimeClass` and use it as described in the [containerd RuntimeClasses documentation](https://github.com/containerd/containerd/blob/main/docs/cri/config.md#runtime-classes)

### CODE REVIEW

**Review by mboersma — COMMENTED** [review]
/ok-to-test

**Comment by mboersma** [comment]
>  To run them the flag `containerd_gvisor_runtime` must be set to `true`.
> Then the user must create a RuntimeClass and use it

Should we (briefly) document these steps somewhere in image-builder itself?

**Comment by heytrav** [comment]
> > To run them the flag `containerd_gvisor_runtime` must be set to `true`.
> > Then the user must create a RuntimeClass and use it
> 
> Should we (briefly) document these steps somewhere in image-builder itself?

Yes I think that would be helpful. I'll have  a look and see where I can add something but I'm happy to add it wherever the preferred location is.

**Comment by mboersma** [comment]
> wherever the preferred location is.

I looked at the [Image Builder book](https://image-builder.sigs.k8s.io/) and I think we could either add something to the [main page](https://github.com/kubernetes-sigs/image-builder/blob/main/docs/book/src/capi/capi.md) or create a new topic like "Customizing containerd".

**Review by mboersma — COMMENTED** [review]
/lgtm
/assign @AverageMarcus 

The docs you added are 💯, thanks so much!

File: images/capi/ansible/roles/containerd/tasks/main.yml [hunk 1]
```diff
@@ -177,3 +177,26 @@
   file:
     path: /tmp/containerd_wasm_shims.tar.gz
     state: absent
+
+- name: Download runsc for gvisor
+  ansible.builtin.get_url:
+    dest: "{{ sysusr_prefix }}/bin/{{ item }}"
+    url: "https://storage.googleapis.com/gvisor/releases/release/latest/{{ ansible_architecture }}/{{ item }}"
```
Comment (AverageMarcus) [inline images/capi/ansible/roles/containerd/tasks/main.yml:184]: Is it safe / desirable to always have latest? Would we want to override it at all?

File: docs/book/src/capi/containerd/customizing-containerd.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+## Running sandboxed containers using gVisor
+
+For additional security in a Kubernetes cluster it can be useful to run certain
+containers in a restricted runtime environment known as a sandbox. One option for this
+is to use [gVisor](https://gvisor.dev/docs/) which provides a layer
+of separation between a running container and the host kernel.
+
+To use gVisor, the necessary executables and [containerd configuration](https://github.com/containerd/containerd/blob/main/docs/cri/config.md#runtime-classes) can be added
+to the Kubernetes golden image during the build by setting the `containerd_gvisor_runtime`
```
Comment (AverageMarcus) [inline docs/book/src/capi/containerd/customizing-containerd.md:11]: I'm not really sure what the term "Kubernetes golden image" means. Do we simply mean the output from image-builder? If so can we say that so it doesn't sound like this is something else the user needs to be aware of.

**Review by AverageMarcus — COMMENTED** [review]
This is great! Thanks @heytrav!

Just a couple very small comments I'm not sure of. 😄

File: docs/book/src/capi/containerd/customizing-containerd.md (hunk 2 above)
Comment (heytrav) [reply docs/book/src/capi/containerd/customizing-containerd.md]: Ok I've updated that.

File: images/capi/ansible/roles/containerd/tasks/main.yml (hunk 1 above)
Comment (AverageMarcus) [reply images/capi/ansible/roles/containerd/tasks/main.yml]: That sounds good to me if the URL format supports that :)

File: docs/book/src/capi/containerd/customizing-containerd.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+## Running sandboxed containers using gVisor
+
+For additional security in a Kubernetes cluster it can be useful to run certain
+containers in a restricted runtime environment known as a sandbox. One option for this
+is to use [gVisor](https://gvisor.dev/docs/) which provides a layer
+of separation between a running container and the host kernel.
+
+To use gVisor, the necessary executables and [containerd configuration](https://github.com/containerd/containerd/blob/main/docs/cri/config.md#runtime-classes) can be added
+to the image generated by image image_builder by setting the `containerd_gvisor_runtime`
```
Comment (AverageMarcus) [inline docs/book/src/capi/containerd/customizing-containerd.md:11]:
```suggestion
to the image generated with image-builder by setting the `containerd_gvisor_runtime`
```

File: images/capi/ansible/roles/containerd/tasks/main.yml (hunk 1 above)
Comment (AverageMarcus) [reply images/capi/ansible/roles/containerd/tasks/main.yml]: And if it doesn't I'm happy to just stick with latest for now.

Comment (heytrav) [reply images/capi/ansible/roles/containerd/tasks/main.yml]: According to the docs it should be possible to use `yyyymmmdd` formatted dates corresponding to releases so I've added that as an option.

**Pull request merged**